geanyprj/src/utils.c

```diff
@@ -275,3 +275,13 @@ GSList *get_file_list(const gchar *path, guint * length, gboolean(*func)(const g
 	return list;
 }
 
+/* update Geany's non-filetype working directories in the build menu to wdir */
+void set_non_ft_build_wdir(gchar* wdir) {
+	gint i;
+	for(i = 0; i < build_get_group_count(GEANY_GBG_NON_FT); ++i)
+		build_set_menu_item(GEANY_BCS_PREF, /* medium priority */
+							GEANY_GBG_NON_FT, /* non-filetype build entries */
+							i,
+							GEANY_BC_WORKING_DIR, /* working dir field */
+							wdir);
```

Comment:
braces around the `for` would be nice for readability IMO as the body spans multiple lines (even though it's a single statement so doesn't change meaning)